src/Dropdown/Dropdown.tsx

```diff
@@ -48,46 +46,52 @@ export interface DropdownProps extends BlockProps {
   useTriggerComponentWidth?: boolean
   onOutsideClick?: (...rest: any) => void
   test?: string
+  makeCSS?: (position: Positioning, targetDimensions: Dimensions, parentDimensions: Dimensions) => string
 }
 
 export const Dropdown: React.FC<DropdownProps> = ({
   className,
   triggerComponent,
   children,
   open,
-  horizontal = 'auto',
-  vertical = 'auto',
+  horizontal = Horizontal.AUTO,
+  vertical = Vertical.AUTO,
   breakDistance = 20,
   useTriggerComponentWidth,
   onOutsideClick,
   test = 'dropdown',
+  makeCSS = makeDefaultCSS,
   ...blockProps
 }) => {
-  const [refs, layout, handleLayout] = useSnapEdgeToParent(breakDistance, { horizontal, vertical })
+  const [{ target , parent }, positioning, handleLayout] = useSnapEdgeToParent(breakDistance, { horizontal, vertical })
 
   useEffect(() => {
     window.addEventListener('resize', handleLayout)
     return () => window.removeEventListener('resize', handleLayout)
   }, [])
 
   const position = useMemo(() => {
-    if (refs.parent.current) {
-      return makeCssPosition(layout, refs.parent.current.getBoundingClientRect().height)
+    if (parent.rect && target.rect) {
+      return makeCSS(
+        positioning,
+        { height: target.rect.height, width: target.rect.width },
+        { height: parent.rect.height, width: parent.rect.width }
```

Comment:
Maybe a tiny helper function called getDimensions would be nice here so it becomes 
```typescript
makeCSS(
    positioning,
    getDimensions(target.rect),
    getDimensions(parent.rect)
)
```

Reply:
I think it would be even better to first of all pass in the entire rect instead of only the height & width and second of all to pass it in the same object.
```
makeCSS(
    positioning,
    { content: target.rect, trigger: parent.rect }
)
```

Reply:
@Stefanwullems is it that means you can use any property that getBoundingClientRect provide in CSS?

Reply:
Yeah exactly!